assets/js/api.js

```diff
@@ -1,644 +1,368 @@
-          var marker = L.circleMarker(latlng, geojsonMarkerOptions);
-          if (typeof feature.properties.issuedate !== 'undefined') {
-            dataType = "Permit";
-            var date = feature.properties.issuedate;
-            var value = formatCurrency(feature.properties.value);
-          } else {
-            dataType = "Demolition";
-            var date = feature.properties.demolition_date;
-            var value = "NA";
-          }
-          var popupContent = "<br><strong>" + dataType + "</strong> "
-                              + "<br><strong>Feature ID:</strong> "
-                              + String(feature.properties.id) + "<hr>"
-                              + "<strong>Address: </strong>"
-                              + feature.properties.address
-                              + "<br><strong>Units: </strong>"
-                              + String(feature.properties.units)
-                              + "<br><strong>Date: </strong>"
-                              + String(date)
-                              + "<br><strong>Size: </strong>"
-                              + String(feature.properties.sqft) + " sqft"
-                              + "<br><strong>Value: </strong>"
-                              + String(value);
-
-          marker.bindPopup(popupContent);
-          return marker;
-        },
-        style: function(data) {
-          //return markerStyle(data);
-            var year = parseInt(data.properties['start']);//.get('year'));
-            // console.log(data.properties.issuedate);
-            if (typeof data.properties.issuedate !== 'undefined') {
-              // console.log('not undefined');
-              dataType = "permits";
-              var color = getPermitColor(2015);//getPermitColor(year);
-            } else {
-              dataType = "demolitions";
-              var color = getDemolitionsColor(2015);//getDemolitionsColor(year);
-            }
-
-            return {
-              radius: 8,
-              fillColor: color,//getPermitColor(year, dataType),//'#FF5500',
-              color: '#000',
-              weight: 1,
-              opacity: 1,
-              fillOpacity: 0.7
-            };
-          }
+    // Should refactor this! Maybe attach a data attribute to each checkbox,
+    // then iterate through those needing to be added to the animation and
+    // call get permits for each of them, then apply 'then' callback
+    if (needPermits && needDemolitions) {
+      $.when(
+        // if we need to make sure permits finishes before demolitions b/c of the date
+        // range, can chain another "then" with the demolitions call; but we should
+        // instead constrain the date selection to the data availability
+        getPermits(yearStart + '-01-01', yearEnd + '-12-31', nbhoodVal, "residential", "permits"),
+        getPermits(yearStart + '-01-01', yearEnd + '-12-31', nbhoodVal, "residential", "demolitions")
+      ).then(function() {
```

Comment:
These 9 lines of code defining the 'then' callback function, look the same for each of the 3 queries related to permits and demolitions. It would be better to define that function once and then reference it in the 3 then clauses.